pvlib/iam.py

```diff
@@ -527,3 +528,221 @@ def sapm(aoi, module, upper=None):
         iam = pd.Series(iam, aoi.index)
 
     return iam
+
+
+def marion_diffuse(model, surface_tilt, regions=None, **kwargs):
+    """
+    Determine diffuse irradiance incidence angle modifiers using Marion's
+    method of integrating over solid angle.
+
+    Parameters
+    ----------
+    model : str
+        The IAM function to evaluate across solid angle.  Must be one of
+        `'ashrae', 'physical', 'martin_ruiz', 'sapm'`.
+
+    surface_tilt : numeric
+        Surface tilt angles in decimal degrees.
+        The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90).
+
+    regions : list, default ['sky', 'horizon', 'ground']
```

Comment:
Is the rationale for `None` to avoid mutable defaults? If so, how about an immutable tuple `('sky', 'horizon', 'ground')`? That would be more clear to me.

Reply:
I'd also consider removing the kwarg entirely so the function always returns all three components. Users that want a single component can use the lower level function.